extend hash interface in hdk (#1222)
holochain/holochain: crates/holochain_zome_types/CHANGELOG.md

## [Unreleased](https://github.com/holochain/holochain/holochain_zome_types-v0.0.2-alpha.1...HEAD)

> - BREAKING CHANGE: Many hashing algorithms can now be specified although none of them do anything yet [#1222](https://github.com/holochain/holochain/pull/1222)

Reviewer: ```suggestion
- BREAKING CHANGE: Many hashing algorithms can now be specified although only the `Entry` hash type does anything yet [#1222](https://github.com/holochain/holochain/pull/1222)
```
